src/runtime/hexagon/fastrpc/tvm_hexagon_remote.h

```diff
@@ -0,0 +1,138 @@
+ * made.
+ * In any case, this submission is "Not a Contribution" with respect to its
+ * permitted use with any of the "vta" and "verilog" subdirectories in the TVM
+ * repo.
+ */
+#ifndef TVM_RUNTIME_HEXAGON_FASTRPC_TVM_HEXAGON_REMOTE_H_
+#define TVM_RUNTIME_HEXAGON_FASTRPC_TVM_HEXAGON_REMOTE_H_
+/// @file tvm_hexagon_remote.idl
+///
+// qidl copyright
+// qidl nested=false
+#include "AEEStdDef.h"
+#include "android_Release_aarch64/remote.h"
+#ifndef __QAIC_HEADER
+#define __QAIC_HEADER(ff) ff
+#endif  // __QAIC_HEADER
+
+#ifndef __QAIC_HEADER_EXPORT
+#define __QAIC_HEADER_EXPORT
+#endif  // __QAIC_HEADER_EXPORT
+
+#ifndef __QAIC_HEADER_ATTRIBUTE
+#define __QAIC_HEADER_ATTRIBUTE
+#endif  // __QAIC_HEADER_ATTRIBUTE
+
+#ifndef __QAIC_IMPL
+#define __QAIC_IMPL(ff) ff
+#endif  // __QAIC_IMPL
+
+#ifndef __QAIC_IMPL_EXPORT
+#define __QAIC_IMPL_EXPORT
+#endif  // __QAIC_IMPL_EXPORT
+
+#ifndef __QAIC_IMPL_ATTRIBUTE
+#define __QAIC_IMPL_ATTRIBUTE
+#endif  // __QAIC_IMPL_ATTRIBUTE
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * Opens the handle in the specified domain.  If this is the first
+ * handle, this creates the session.  Typically this means opening
+ * the device, aka open("/dev/adsprpc-smd"), then calling ioctl
+ * device APIs to create a PD on the DSP to execute our code in,
+ * then asking that PD to dlopen the .so and dlsym the skel function.
+ *
+ * @param uri, <interface>_URI"&_dom=aDSP"
+ *    <interface>_URI is a QAIC generated uri, or
+ *    "file:///<sofilename>?<interface>_skel_handle_invoke&_modver=1.0"
+ *    If the _dom parameter is not present, _dom=DEFAULT is assumed
+ *    but not forwarded.
+ *    Reserved uri keys:
+ *      [0]: first unamed argument is the skel invoke function
+ *      _dom: execution domain name, _dom=mDSP/aDSP/DEFAULT
+ *      _modver: module version, _modver=1.0
+ *      _*: any other key name starting with an _ is reserved
+ *    Unknown uri keys/values are forwarded as is.
+ * @param h, resulting handle
+ * @retval, 0 on success
+ */
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_open)(
+    const char* uri, remote_handle64* h) __QAIC_HEADER_ATTRIBUTE;
+/**
+    * Closes a handle.  If this is the last handle to close, the session
+    * is closed as well, releasing all the allocated resources.
+
+    * @param h, the handle to close
+    * @retval, 0 on success, should always succeed
+    */
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_close)(
+    remote_handle64 h) __QAIC_HEADER_ATTRIBUTE;
+typedef struct _tvm_hexagon_remote_buffer__seq_octet
+    _tvm_hexagon_remote_buffer__seq_octet;
+typedef _tvm_hexagon_remote_buffer__seq_octet tvm_hexagon_remote_buffer;
+struct _tvm_hexagon_remote_buffer__seq_octet {
+  unsigned char* data;
+  int dataLen;
+};
+typedef unsigned int tvm_hexagon_remote_handle_t;
+typedef uint64 tvm_hexagon_remote_scalar_t;
+__QAIC_HEADER_EXPORT int __QAIC_HEADER(tvm_hexagon_remote_map_buffer_to_dsp)(
```

Comment:
could we also upload `tvm_hexagon_remote_map_buffer_to_dsp` definition? i.e. `tvm_hexagon_remote_map_buffer_to_dsp_stub.c` / `tvm_hexagon_remote_map_buffer_to_dsp_skel.c` / tvm_hexagon_remote_map_buffer_to_dsp.il`